repository/git.go

```diff
@@ -745,6 +946,29 @@ func (repo *GitRepo) ListNotedRevisions(notesRef string) []string {
 	return revisions
 }
 
+// Remotes returns a list of the remotes.
+func (repo *GitRepo) Remotes() ([]string, error) {
+	remotes, err := repo.runGitCommand("remote")
+	if err != nil {
+		return nil, err
+	}
+	remoteNames := strings.Split(remotes, "\n")
+	var result []string
+	for _, name := range remoteNames {
+		result = append(result, strings.TrimSpace(name))
+	}
+	sort.Strings(result)
+	return result, nil
+}
+
+// Fetch fetches from the given remote using the supplied refspecs.
+func (repo *GitRepo) Fetch(remote string, fetchSpecs []string) error {
```

Comment:
Nit: maybe `fetchSpecs` should be renamed `refspec` to better align with the git CLI, or if it is intended to be something different, than maybe it should be more opinionated than `[]string` (although I think this is just refspec...)

Reply:
Done